tests/unit_tests/test_coverage.py

```diff
@@ -34,21 +89,116 @@ def test_coverage_get_from_spades(metagenome, tmp_path):
     assert out.exists()
 
 
-@pytest.mark.skip
-@pytest.mark.wip
-def test_coverage_get_from_sam(metagenome, tmp_path):
+def test_coverage_get_from_sam(metagenome, sam_alignment, tmp_path):
     out = tmp_path / "covs_from_sam.tsv"
-    df = coverage.get(fasta=metagenome, from_spades=False, out=out)
+
+    df = coverage.get(fasta=metagenome, from_spades=False, out=out, sam=sam_alignment)
     assert df.index.name == "contig"
     assert "coverage" in df.columns
     assert out.exists()
 
 
-@pytest.mark.skip
-@pytest.mark.wip
-def test_coverage_get_from_bam(metagenome, tmp_path):
+def test_coverage_get_from_bam(metagenome, bam_alignment, tmp_path):
     out = tmp_path / "covs_from_bam.tsv"
-    df = coverage.get(fasta=metagenome, from_spades=False, out=out)
+    df = coverage.get(fasta=metagenome, from_spades=False, out=out, bam=bam_alignment)
     assert df.index.name == "contig"
     assert "coverage" in df.columns
     assert out.exists()
+
+
+def test_coverage_get_from_bed(metagenome, bed_alignment, tmp_path):
+    out = tmp_path / "covs_from_bed.tsv"
+    df = coverage.get(
+        fasta=metagenome, from_spades=False, out=out.as_posix(), bed=bed_alignment
+    )
+    assert df.index.name == "contig"
+    assert "coverage" in df.columns
+    assert out.exists()
+
+
+def test_coverage_get_from_reads(metagenome, fwd_reads, rev_reads, tmp_path):
+    out = tmp_path / "covs_from_bed.tsv"
+    df = coverage.get(
+        fasta=metagenome,
+        from_spades=False,
+        out=out.as_posix(),
```

Comment:
Looking into Pathlib methods/operators [:link:](https://docs.python.org/3/library/pathlib.html#operators "Pathlib operators") it seems we should be using `str(out)` here instead of `out.as_posix()`

```suggestion
        out=str(out),
```

Reply:
It seems we don't asctually need `str` as just inputing the `posix path` is working as well.

Reply:
Also, I found a [good resource](https://www.linuxjournal.com/content/easier-python-paths-pathlib) on why using pathlib objects is actually a more python way of handling files.